scripts/push-docker-image.sh

```diff
@@ -1,31 +1,71 @@
 #!/usr/bin/env bash
+#
+# Script is meant to build multi-arch container images and publish them to multiple container registries
+# 
+# Script is:
+# - figuring out if an image is a development one (by default suffixed with `-dev` in image name)
+# - figuring out the image tag (aka version) based on GITHUB_REF value
+#
+# Script is not:
+# - directly executing `docker build`. This is done in Makefile
+# - logging to registries 
+#
+
 # exit immediately when a command fails
 set -e
 # only exit with zero if all commands of the pipeline exit successfully
 set -o pipefail
-# error on unset variables
-set -u
```

Comment:
`GITHUB_REF` is not set when using the script locally, hence I deleted this check.